sdcm/utils/k8s.py

```diff
@@ -609,6 +614,251 @@ def stop(self, timeout=None):
+        #          "name":"sct-cluster-dc-1-kind-0",
+        #          "namespace": "scylla",
+        #          ...
+        #      },
+        #      "subsets": [{
+        #          "addresses": [
+        #              {"ip": "10.244.218.195", ... },
+        #              ...
+        #          ],
+        #          "ports": [...]
+        #      }]
+        #  }
+        # }
+        data = {}
+        try:
+            LOGGER.debug("Processing following line: %s", line)
+            data = yaml.safe_load(line) or {}
+            metadata = data.get('object', {}).get('metadata', {})
+            namespace = metadata.get('namespace')
+            if not namespace:
+                raise KeyError("Cannot find 'namespace' in the line: %s" % line)
+            if namespace not in self.mapper_dict:
+                self.mapper_dict[namespace] = {}
+            # NOTE: following condition allows to skip processing of all the
+            #       load-balancer/headless endpoints that do not refer to any Scylla pod.
+            labels = metadata.get('labels', {})
+            if not all((key in labels.keys() for key in self.SCYLLA_PODS_EXPECTED_LABEL_KEYS)):
+                return None
+
+            if name := metadata.get('name'):
+                if name not in self.mapper_dict[namespace]:
+                    self.mapper_dict[namespace][name] = {}
+                if 'current_ip' not in self.mapper_dict[namespace][name]:
+                    self.mapper_dict[namespace][name]['current_ip'] = None
+                if 'old_ips' not in self.mapper_dict[namespace][name]:
+                    self.mapper_dict[namespace][name]['old_ips'] = []
+            for subset in data.get('object', {}).get('subsets', []):
+                for address in subset.get('addresses', []):
+                    current_ip_address = address['ip']
+                    if current_ip_address == self.mapper_dict[namespace][name]['current_ip']:
+                        continue
+                    old_ip_candidate = self.mapper_dict[namespace][name]['current_ip']
+                    self.mapper_dict[namespace][name]['current_ip'] = current_ip_address
+                    if not old_ip_candidate:
+                        break
+                    self.mapper_dict[namespace][name]['old_ips'].append(old_ip_candidate)
+                    LOGGER.info(
+                        "'%s/%s' node has changed it's pod IP address from '%s' to '%s'",
+                        namespace, name, old_ip_candidate, current_ip_address)
+                    self._call_callbacks(namespace, name)
+                    break
+                else:
+                    break
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.warning(
+                "Failed to parse following line: %s\nerr: %s", line, exc)
+
+    def _call_callbacks(self, namespace, pod_name) -> None:
+        # TODO: run callbacks in parallel on per-namespace basis and serially inside a
```

Comment:
how much time our callback takes ? 
refresh_ip shouldn't be that long, and update ip in monitor should be too long as well, i.e. it's less then a few seconds, long before the scylla process is ready

Reply:
It takes seconds. But this todo is not urgent, because workability of the feature as a whole is much more needed.